cgmes/cgmes-measurements/src/main/java/com/powsybl/cgmes/measurements/CgmesAnalogPostProcessor.java

```diff
@@ -0,0 +1,141 @@
+            Identifiable<?> identifiable = network.getIdentifiable(terminalId);
+            if (identifiable != null) {
+                createMeas(identifiable, id, terminalId, measurementType, typesMapping);
+                return;
+            }
+            LOG.warn("Ignored terminal {} of {} {}: not found", terminalId, measurementType, id);
+        }
+        Identifiable<?> identifiable = network.getIdentifiable(powerSystemResourceId);
+        if (identifiable != null) {
+            createMeas(identifiable, id, terminalId, measurementType, typesMapping);
+            return;
+        }
+        PropertyBag bay = bays.stream().filter(b -> b.getId("Bay").equals(powerSystemResourceId)).findFirst().orElse(null);
+        if (bay != null) {
+            String voltageLevelId = bay.getId("VoltageLevel");
+            LOG.info("Power resource system {} of Analog {} is a Bay: Analog is attached to the associated voltage level {}",
+                    powerSystemResourceId, id, voltageLevelId);
+            VoltageLevel voltageLevel = network.getVoltageLevel(voltageLevelId);
+            if (voltageLevel == null) {
+                LOG.warn("Ignored {} {}: associated voltage level {} not found", measurementType, id, voltageLevelId);
+                return;
+            }
+            voltageLevel.setProperty(Conversion.CGMES_PREFIX_ALIAS_PROPERTIES + "Analog_" + measurementType, id);
+        } else {
+            LOG.warn("Ignored {} {}: attached power system resource {} not found", measurementType, id, powerSystemResourceId);
+        }
+    }
+
+    private static void createMeas(Identifiable<?> identifiable, String id, String terminalId, String measurementType, Map<String, String> typesMapping) {
+        if (identifiable instanceof Connectable) {
+            Connectable<?> c = (Connectable<?>) identifiable;
+            Measurements meas = c.getExtension(Measurements.class);
+            if (meas == null) {
+                c.newExtension(MeasurementsAdder.class).add();
+                meas = c.getExtension(Measurements.class);
+            }
+            Measurement.Type type = getType(measurementType, typesMapping);
+            Measurement.Side side = null;
+            MeasurementAdder adder = meas.newMeasurement()
+                    .setValid(false)
+                    .setId(id);
+            if (!(c instanceof Injection)) {
+                side = getSide(terminalId, c);
+            }
+            if (type != OTHER && side == null && !(c instanceof Injection)) {
+                adder.setType(OTHER);
+            } else {
+                adder.setType(type);
+            }
+            adder.setSide(side);
+            Measurement measurement = adder.add();
+            measurement.putProperty("cgmesType", measurementType);
+        } else {
+            identifiable.setProperty(Conversion.CGMES_PREFIX_ALIAS_PROPERTIES + "Analog_" + measurementType, id);
+        }
+        // TODO get value of measurement
```

Comment:
You write a todo here because it is complex to do?

Reply:
Yes, and since the IDs were asked first, this allows to do a first approach